trunk/NDHMS/Routing/Reservoirs/reservoir_tests.F

```diff
@@ -33,4 +47,79 @@ function test_levelpool() result(rv)
 
     end function test_levelpool
 
+
+    ! This tests the reservoir function of the level pool module under the specific condition
+    ! where the water elevation reaches the max height.
+    function test_levelpool_overflow_max_height() result(rv)
+
+        implicit none
+        logical rv                       ! test result
+        type (levelpool_struct) :: levelpool_reservoir_data
+        real :: lake_area, weir_elevation, weir_coeffecient
+        real :: weir_length, orifice_elevation, orifice_coefficient
+        real :: orifice_area, max_depth
+        integer :: lake_number
+        real :: inflow, prev_time_inflow, outflow, water_elevation
+        real, dimension(108) :: inflow_array
+        integer :: timestep_count
+
+        rv = .false.
+        prev_time_inflow = 0.0
+        timestep_count = 0
+        water_elevation = 0.0
+
+        lake_area = 1.509490013122558594e+01
+        weir_elevation = 9.626000022888183238e+00
+        weir_coeffecient = 4.000000000000000222e-01
```

Comment:
why is weir_coefficient not = 0.4.
Also coefficient is spelled incorrectly.

Reply:
@jdmattern-noaa fixed both